service/index/reader.go

```diff
@@ -22,23 +22,24 @@ import (
 
 	"github.com/onflow/flow-go/ledger"
 	"github.com/onflow/flow-go/model/flow"
-
-	"github.com/optakt/flow-dps/service/storage"
 )
 
 // Reader implements the `index.Reader` interface on top of the DPS server's
 // Badger database index.
 type Reader struct {
+	storage StorageLibrary
```

Comment:
Do we really need interfaces with two words in their name?

```go
type Reader struct {
  db      *badger.DB
  storage Storage
}
```

...looks fine to me.